docs/api/generator/ecr.md

```diff
@@ -0,0 +1,26 @@
+ECRAuthorizationTokenSpec uses the GetAuthorizationToken API to retrieve an authorization token.
```

Comment:
General naive question here - Is there a way to keep track on the generated tokens? Are they bound to any costs?

Reply:
> Is there a way to keep track on the generated tokens

No, there is no way to keep track of the generated tokens from a `generator` point of view. I think there are use-cases for it, but for now the only use-case i've stumbled upon is vault with it's various secret backends and the ability to renew/revoke the lease of a token. (I still have the approach from #1211 in the back of my head on how we could tackle that).

Conceptually, i'd say generators shouldn't care about that state at all. Instead the responsibility should be pushed to the user. (choose a `refreshInterval` that suites the needs).

> Are they bound to any costs?

Specifically for ECR: no, there are none. They're generated and valid for a static duration, that's it.

Reply:
> No, there is no way to keep track of the generated tokens from a generator point of view. I think there are use-cases for it, but for now the only use-case i've stumbled upon is vault with it's various secret backends and the ability to renew/revoke the lease of a token. (I still have the approach from https://github.com/external-secrets/external-secrets/pull/1211 in the back of my head on how we could tackle that).

No issues with that! I actually asked that because I've let the code run with `RefreshInterval:10s` for a couple of minutes, and generated hundreds of tokens lol. Just wanted to know if there was a way to revoke them in AWS 😛